graylog2-web-interface/src/views/components/widgets/Widget.test.jsx

```diff
@@ -106,55 +170,100 @@ describe('<Widget />', () => {
 
-    const cancelButton = wrapper.find('button[children="Cancel"]');
-    cancelButton.simulate('click');
+    const cancelButton = getByText('Cancel');
+    fireEvent.click(cancelButton);
 
     expect(WidgetActions.updateConfig).toHaveBeenCalledWith('widgetId', { foo: 42 });
   });
   it('does not restores original state of widget config when clicking "Finish Editing"', () => {
     const widgetWithConfig = { config: { foo: 42 }, id: 'widgetId', type: 'dummy' };
-    const wrapper = mount(<DummyWidget editing widget={widgetWithConfig} />);
+    const { getByText } = render(<DummyWidget editing widget={widgetWithConfig} />);
 
-    const editComponent = wrapper.find('edit-dummy-visualization').at(0);
-    const widgetConfigChange = editComponent.props().onChange;
-
-    WidgetActions.updateConfig = mockAction(jest.fn());
-    widgetConfigChange({ foo: 23 });
+    const onChangeBtn = getByText('Click me');
+    fireEvent.click(onChangeBtn);
     expect(WidgetActions.updateConfig).toHaveBeenCalledWith('widgetId', { foo: 23 });
 
-    const cancelButton = wrapper.find('button[children="Save"]');
-    cancelButton.simulate('click');
+    const saveButton = getByText('Save');
+    fireEvent.click(saveButton);
 
     expect(WidgetActions.updateConfig).not.toHaveBeenCalledWith('widgetId', { foo: 42 });
   });
+
+  describe('copy widget to dashboard', () => {
+    beforeEach(() => {
+      DashboardsStore.getInitialState = jest.fn(() => dashboardState);
+      ViewStore.getInitialState = jest.fn(() => viewStoreState);
+      ViewManagementActions.get = mockAction(jest.fn((() => Promise.resolve(dashboard1.toJSON()))));
+      SearchActions.get = mockAction(jest.fn(() => Promise.resolve(searchDB1.toJSON())));
+      ViewManagementActions.update = mockAction(jest.fn(() => Promise.resolve()));
+      SearchActions.create = mockAction(jest.fn(() => Promise.resolve({ search: searchDB1.toJSON() })));
+      Routes.pluginRoute = jest.fn(route => id => `${route}-${id}`);
+      browserHistory.push = jest.fn();
+      asMock(CopyWidgetToDashboard).mockImplementation(() => View.builder()
+        .search(Search.builder().id('search-id').build())
+        .id('new-id').type(View.Type.Dashboard)
+        .build());
+    });
+    afterEach(() => {
+      cleanup();
+    });
+    const renderAndClick = () => {
+      const { getByText, getByTestId } = render(<DummyWidget />);
+      const actionToggle = getByTestId('widgetActionDropDown');
+      fireEvent.click(actionToggle);
+      const copyToDashboard = getByText('Copy to Dashboard');
+      fireEvent.click(copyToDashboard);
+      const view1ListItem = getByText('view 1');
+      fireEvent.click(view1ListItem);
+      const selectBtn = getByText('Select');
+      fireEvent.click(selectBtn);
+    };
+
+    it('should get dashboard from backend', async () => {
+      renderAndClick();
+      await wait(() => expect(ViewManagementActions.get).toHaveBeenCalledTimes(1));
```

Comment:
We should also check the parameters of the action call here and the following tests.